independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanGenerator.java

```diff
@@ -1089,6 +1091,46 @@ private void implementCreateForSyntheticBean(ClassCreator beanCreator, BeanInfo
                 .setModifiers(ACC_PRIVATE);
         bean.getCreatorConsumer().accept(createSynthetic);
 
+        // TODO move below, after `SyntheticCreationalContextImpl` is created, and pass it to this method
```

Comment:
Adding a comment here so we don't forget to handle the TODO.

Reply:
FTR, this TODO is wrong and I'm removing it from the code. This is related to the other thread, where we discussed whether the active check procedure should be `Function<SyntheticCreationalContext, ActiveResult>` and ended up deciding that it shouldn't. (It might end up being `Function<ActiveCheckContext, ActiveResult>` instead, but the TODO still wouldn't apply.)